crates/oxc_linter/src/rules/no_duplicate_case.rs

```diff
@@ -0,0 +1,173 @@
+use std::collections::HashSet;
+
+use oxc_ast::{AstKind, Span};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+
+use crate::{ast_util::calculate_hash, context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(no-duplicate-case): Disallow duplicate case labels")]
+#[diagnostic(severity(warning), help("Remove the duplicated case"))]
+struct NoDuplicateCaseDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoDuplicateCase;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Disallow duplicate case labels
+    ///
+    /// ### Why is this bad?
+    ///
+    /// If a switch statement has duplicate test expressions in case clauses,
+    /// it is likely that a programmer copied a case clause but forgot to change the test expression.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// var a = 1;
+    /// switch (a) {
+    ///     case 1:
+    ///         break;
+    ///     case 1:
+    ///         break;
+    ///     case 2:
+    ///         break;
+    ///     default:
+    ///         break;
+    /// }
+    /// ```
+    NoDuplicateCase,
+    correctness
+);
+
+impl Rule for NoDuplicateCase {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::SwitchStatement(ss) = node.get().kind() {
+            let mut set = HashSet::new();
```

Comment:
```
        let mut set: FxHashMap<u64, Span> = FxHashMap::default();
        set.reserve(ss.cases.len());
```

Reply:
Why using `HashMap` instead of `HashSet` here. Because we need to print out all the duplicated cases?

Reply:
refactored in [94ed85d](https://github.com/Boshen/oxc/pull/179/commits/94ed85d36766c864eba57f1f1eed76586b68cb97)

Reply:
yes, you should label all the duplicate cases. Point to two of them. I don't know if we can point to a vec of spans (probably not).

Reply:
It seems that we cannot specify a vec in `NoDuplicateCaseDiagnostic` because the `Vec<Span>` doesn't implement the `SourceSpan`.